l10n_us_check_writing_address/report/print_check.py

```diff
@@ -0,0 +1,135 @@
+            first page is valid. This function returns a list of stub lines per page.
+        """
+        if len(payment.invoice_ids) == 0:
+            return None
+
+        multi_stub = payment.company_id.us_check_multi_stub
+
+        invoices = payment.invoice_ids.sorted(key=lambda r: r.date_due)
+        debits = invoices.filtered(lambda r: r.type == 'in_invoice')
+        credits = invoices.filtered(lambda r: r.type == 'in_refund')
+
+        # Prepare the stub lines
+        if not credits:
+            stub_lines = [self.make_stub_line(payment, inv) for inv in invoices]
+        else:
+            stub_lines = [{'header': True, 'name': "Bills"}]
+            stub_lines += [self.make_stub_line(payment, inv) for inv in debits]
+            stub_lines += [{'header': True, 'name': "Refunds"}]
+            stub_lines += [self.make_stub_line(payment, inv) for inv in credits]
+
+        # Crop the stub lines or split them on multiple pages
+        if not multi_stub:
+            # If we need to crop the stub, leave place for an ellipsis line
+            num_stub_lines = len(stub_lines) > INV_LINES_PER_STUB and INV_LINES_PER_STUB-1 or INV_LINES_PER_STUB
+            stub_pages = [stub_lines[:num_stub_lines]]
+        else:
+            stub_pages = []
+            i = 0
+            while i < len(stub_lines):
+                # Make sure we don't start the credit section at the end of a page
+                if len(stub_lines) >= i+INV_LINES_PER_STUB and stub_lines[i+INV_LINES_PER_STUB-1].get('header'):
+                    num_stub_lines = INV_LINES_PER_STUB-1 or INV_LINES_PER_STUB
+                else:
+                    num_stub_lines = INV_LINES_PER_STUB
+                stub_pages.append(stub_lines[i:i+num_stub_lines])
+                i += num_stub_lines
+
+        return stub_pages
+
+    def make_stub_line(self, payment, invoice):
+        """ Return the dict used to display an invoice/refund in the stub
+        """
+        # Find the account.partial.reconcile which are common to the invoice and the payment
+        if invoice.type in ['in_invoice', 'out_refund']:
+            invoice_sign = 1
+            invoice_payment_reconcile = invoice.move_id.line_ids.mapped('matched_debit_ids').filtered(lambda r: r.debit_move_id in payment.move_line_ids)
+        else:
+            invoice_sign = -1
+            invoice_payment_reconcile = invoice.move_id.line_ids.mapped('matched_credit_ids').filtered(lambda r: r.credit_move_id in payment.move_line_ids)
+
+        if payment.currency_id != payment.journal_id.company_id.currency_id:
+            amount_paid = abs(sum(invoice_payment_reconcile.mapped('amount_currency')))
+        else:
+            amount_paid = abs(sum(invoice_payment_reconcile.mapped('amount')))
+
+        return {
+            'due_date': invoice.date_due,
+            'number': invoice.reference and invoice.number + ' - ' + invoice.reference or invoice.number,
```

Comment:
String interpolation is preferred (and more efficient) than concatenation. Also, this might be better as an explicit conditional
```
'%s - %s' % (invoice.number, invoice.reference)
```